### PR TITLE
Apply SELinux label to allocate directory of docker driver

### PR DESCRIPTION
**Pull request by nak3**

### CODE REVIEW

**Comment by nak3** [comment]
Without this fix, driver test (i.e. `$ go test -cover -timeout=500s github.com/hashicorp/nomad/client/drive`) also fails on SELinux enabled environment.

File: client/driver/docker.go [hunk 1]
```diff
@@ -108,8 +108,8 @@ func (d *DockerDriver) containerBinds(alloc *allocdir.AllocDir, task *structs.Ta
 	}
 
 	return []string{
-		fmt.Sprintf("%s:%s", shared, allocdir.SharedAllocName),
-		fmt.Sprintf("%s:%s", local, allocdir.TaskLocal),
+		fmt.Sprintf("%s:%s:rw,Z", shared, allocdir.SharedAllocName),
```
Comment (dadgar) [inline client/driver/docker.go]: The shared alloc is shared across containers so it should be `z` (lowercase)

Comment (dadgar) [inline client/driver/docker.go]: Please put a comment on why these are needed above

**Comment by nak3** [comment]
Thanks, I updated.

File: client/driver/docker.go [hunk 2]
```diff
@@ -108,8 +108,10 @@ func (d *DockerDriver) containerBinds(alloc *allocdir.AllocDir, task *structs.Ta
 	}
 
 	return []string{
-		fmt.Sprintf("%s:%s", shared, allocdir.SharedAllocName),
-		fmt.Sprintf("%s:%s", local, allocdir.TaskLocal),
+		// "z" and "Z" option is to allocate directory with SELinux label.
+		fmt.Sprintf("%s:%s:rw,z", shared, allocdir.SharedAllocName),
+		// capital "Z" will label with Multi-Category Security (MCS) labels
+		fmt.Sprintf("%s:%s:rw,Z", local, allocdir.TaskLocal),
```
Comment (dadgar) [inline client/driver/docker.go]: I pulled your branch and you need to add one more thing for it to work.

Can you change `%s:%s` to `%s:/%s` on both of them. It needs to be an absolute path with that flag.

**Comment by nak3** [comment]
Thanks again. I updated. The test passed both (w/ or w/o `/%s `), so it may need test case.

**Pull request merged**